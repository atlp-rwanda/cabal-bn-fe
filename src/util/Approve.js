import React from 'react';
import { Grid, Button } from '@mui/material';

export default ({ row, handleApprove, handleReject }) => (
  <Grid container justifyContent="center" alignItems="center">
    <Button
      variant="contained"
      disableElevation
      disabled={!!(row?.status === 'APPROVED' || row?.status === 'REJECTED')}
      sx={{
        backgroundColor:
          row?.status === 'APPROVED' || row?.status === 'REJECTED'
            ? '#EC5C5C'
            : '#0ABDA0',
        color: '#fff',
        fontSize: '10px',
        margin: '10px',
        width: '20%',
        '&:hover': {
          backgroundColor: '#0ABDA0',
          color: '#fff',
        },
      }}
      /* istanbul ignore next */
      onClick={() => handleApprove(row.id)}
    >
      Approve
    </Button>
    <Button
      variant="contained"
      disabled={!!(row?.status === 'APPROVED' || row?.status === 'REJECTED')}
      sx={{
        backgroundColor:
          row?.status === 'REJECTED' || row?.status === 'APPROVED'
            ? '#EC5C5C'
            : '#E13535',
        color: '#fff',
        fontSize: '10px',
        width: '20%',
        '&:hover': {
          backgroundColor: '#EC5C5C',
          color: '#fff',
        },
      }}
      /* istanbul ignore next */
      onClick={() => handleReject(row.id)}
    >
      Reject
    </Button>
  </Grid>
);
