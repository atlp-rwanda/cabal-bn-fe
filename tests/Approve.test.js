import React from 'react';
import renderer from 'react-test-renderer';
import Approve from '../src/util/Approve';

describe('Navbar', () => {
  it('should render the spinner on click', () => {
    const component = renderer.create(<Approve />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
      tree.props.close;
    });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
