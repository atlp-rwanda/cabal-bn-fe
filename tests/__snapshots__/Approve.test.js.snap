// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the spinner on click 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1hbmzt3-MuiGrid-root"
>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-5xwig-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Approve
  </button>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-l0xg7d-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Reject
  </button>
</div>
`;

exports[`Navbar should render the spinner on click 2`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1hbmzt3-MuiGrid-root"
>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-5xwig-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Approve
  </button>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-l0xg7d-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Reject
  </button>
</div>
`;
